domain/miningmanager/miningmanager_test.go

```diff
@@ -0,0 +1,446 @@
+func TestInsertDoubleTransactionsToMempool(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestInsertDoubleTransactionsToMempool")
+		if err != nil {
+			t.Fatalf("Error setting up TestConsensus: %+v", err)
+		}
+		defer teardown(false)
+
+		miningFactory := miningmanager.NewFactory()
+		miningManager := miningFactory.NewMiningManager(tc, blockMaxMass, false)
+		transaction := createTransactionWithUTXOEntry(t, params, 0)
+		err = miningManager.ValidateAndInsertTransaction(transaction, true)
+		if err != nil {
+			t.Fatalf("ValidateAndInsertTransaction: %v", err)
+		}
+		err = miningManager.ValidateAndInsertTransaction(transaction, true)
+		if err == nil || !strings.Contains(err.Error(), "already have transaction") {
+			t.Fatalf("ValidateAndInsertTransaction: %v", err)
+		}
+	})
+}
+
+// TestHandleNewBlockTransactions verifies that all the transactions in the block were successfully removed from the memory pool.
+func TestHandleNewBlockTransactions(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestHandleNewBlockTransactions")
+		if err != nil {
+			t.Fatalf("Error setting up TestConsensus: %+v", err)
+		}
+		defer teardown(false)
+
+		miningFactory := miningmanager.NewFactory()
+		miningManager := miningFactory.NewMiningManager(tc, blockMaxMass, false)
+		transactionsToInsert := make([]*externalapi.DomainTransaction, 10)
+		for i := range transactionsToInsert[(coinbaseIndex + 1):] {
+			transaction := createTransactionWithUTXOEntry(t, params, i)
+			transactionsToInsert[i+1] = transaction
+			err = miningManager.ValidateAndInsertTransaction(transaction, true)
+			if err != nil {
+				t.Fatalf("ValidateAndInsertTransaction: %v", err)
+			}
+		}
+
+		_, err = miningManager.HandleNewBlockTransactions(transactionsToInsert[0:3])
+		if err != nil {
+			t.Fatalf("HandleNewBlockTransactions: %v", err)
+		}
+		mempoolTransactions := miningManager.AllTransactions()
+		for _, RemovedTransaction := range transactionsToInsert[(coinbaseIndex + 1):3] {
+			if contains(RemovedTransaction, mempoolTransactions) {
+				t.Fatalf("This transaction shouldnt be in mempool: %v", RemovedTransaction)
+			}
+		}
+
+		// There are no chained/double-spends transactions, and hence it is expected that all the other
+		// transactions, will still be included in the mempool.
+		mempoolTransactions = miningManager.AllTransactions()
+		for i, transaction := range transactionsToInsert[3:] {
+			if !contains(transaction, mempoolTransactions) {
+				t.Fatalf("This transaction %d should be in mempool: %v", i, transaction)
+			}
+		}
+
+		_, err = miningManager.HandleNewBlockTransactions(transactionsToInsert[2:])
```

Comment:
What are you trying to do here?

Reply:
This action should insert all the other transactions into the mempool.
I'll add a comment so it will be more clear.